### DISCUSSION TITLE
Feature Request: Reorder array items between arrays of the same shape

### DISCUSSION THREAD

**Discussion by damnsamn** [payloadcms/payload#4811] [Feature Requests & Ideas]
It would be great to be able to reorder (or otherwise duplicate) array children between parents.

Consider a nested array setup, where each L1 array item expects an array of L2 items:

- L1A
  - L2A
  - L2B
  - L2C
- L1B
  - L2D

There's currently no way to shift, for example, `L2C` into the `L1B` list. The only current way of doing this seems to be creating a new item under `L1B`, copying across all the data from `L2C`, and then deleting `L2C`.

As a real world use case, consider a navigation submenu that is split into columns like this:
<img width="889" alt="image" src="https://github.com/payloadcms/payload/assets/17864782/b8ad8710-ce25-436e-824e-edcec99c3545">

In this case I've made an error: I accidentally put "Brand Design" under "Industries", when really it belongs under "Services". A user might intuitively attempt to drag Brand Design from one array to the other, which would fail. They would have to go through the same `Create New Item > Copy/Paste Data > Delete Old Item` as described above.

I Imagine there'd need to be some sort of immediate validation in place to ensure that the shapes of both arrays are identical in terms of schema.

**Comment by akantic** [comment]
I would love to see this as well. Any chance that this will ever get to the roadmap?
